python/tvm/relay/op/contrib/forge/forge_passes.py

```diff
@@ -3953,7 +3956,183 @@ def callback(self, pre, post, node_map):
                 num_newaxis-=1
             return act
         return post
+    
+class DecomposeGridSample(DFPatternCallback):
+    def __init__(self):
+        super().__init__(rewrite_once=True, require_type=True)
+
+        self.data = wildcard()
+        self.grid = wildcard()
+        self.grid_sample = is_op("image.grid_sample")(self.data, self.grid)
+        self.pattern = self.grid_sample
 
+    def callback(self, pre, post, node_map):
+        from tvm.relay.frontend.common import infer_shape
+        data = node_map[self.data][0]  
+        grid = node_map[self.grid][0]  
+        grid = grid.args[0]  
+
+        mode = post.attrs["method"]
+        align_corners = post.attrs["align_corners"]
+        # Split the image and grid tensors across the batch size 
+        batch_size = infer_shape(data)[0]
+        split_im = tvm.relay.split(data, indices_or_sections=batch_size, axis=0)
```

Comment:
Make sure to explain in funciton description how decomposition works. Please make sure to cover all main cases